packages/core/src/storage.ts

```diff
@@ -208,6 +208,49 @@ export type FindResult<T extends Doc> = WithLookup<T>[] & {
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface TxResult {}
 
+/**
+ * @public
+ */
+export interface IndexedDoc {
+  id: Ref<Doc>
+  _class: Ref<Class<Doc>>
+  space: Ref<Space>
+  modifiedOn: Timestamp
+  modifiedBy: Ref<Account>
+  attachedTo?: Ref<Doc>
+  attachedToClass?: Ref<Class<Doc>>
+  [key: string]: any
+}
+
+/**
+ * @public
+ */
+export interface SearchQuery {
+  query: string
+  aggregateBy?: string
+  filter?: {
+    [key: string]: any
+  }
+}
+
+/**
+ * @public
+ */
+export interface SearchOptions {
+  limit?: number
+  offset?: number
```

Comment:
Why we need offset?

Reply:
Should we have sorting?

Reply:
I put offset there to be able to build pagination for results. I'm not sure that we would need it in mentions or spotlight, but it might be useful if we will want to introduce "show more".

I thought about sorting, but I believe in search we always want to use scoring so we would have a question how to tune the scoring in general and how to make search more relevant.

Reply:
I think we do not need a pagination, assume we have many results, user will type more to find a goal.

Reply:
No problem, I will remove it for now, it's not hard to add it if we need it